web/src/components/storage/VolumeLocationSelectorTable.jsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+// @ts-check
+
+import React from "react";
+import { Chip } from '@patternfly/react-core';
+
+import { _ } from "~/i18n";
+import {
+  DeviceName, DeviceDetails, DeviceSize, toStorageDevice
+} from "~/components/storage/device-utils";
+import { ExpandableSelector } from "~/components/core";
+
+/**
+ * @typedef {import("../core/ExpandableSelector").ExpandableSelectorColumn} ExpandableSelectorColumn
```

Comment:
NP: Did `~/components/core/ExpandableSelector` not work? Or maybe did you though it is too long? By the way, maybe `ExpandableSelectorColumn` can be names just `Column` in `core/ExpandableSelector.jsx` You have to rename it anyway when _importing_ it.

```
 * @typedef {import("~/components/core/ExpandableSelector").Column} ExpandableSelectorColumn
```

Again, not related with this PR but this kind of things makes me tend to think that we are actually running away of TypeScript with no strong reason.

Reply:
Good catch! The path was automatically added by my editor :).